From 916c57dd3fca5c1d53ebcf14db486555f16d9d89 Mon Sep 17 00:00:00 2001
From: Ilya Alekseyev <ilya.alekseyev@nutanix.com>
Date: Wed, 11 Oct 2023 22:07:22 -0400
Subject: [PATCH 05/16] Nutanix improvements

- Fetch Nutanix RHEL source image URL from environment
- Force-delete Nutanix builder VMs on failure
- Always download Nutanix source images and delete them after build
Signed-off-by: Vignesh Goutham Ganesh <vgg@amazon.com>
---
 images/capi/packer/nutanix/packer.json.tmpl | 5 +++--
 images/capi/packer/nutanix/rhel-8.json      | 2 +-
 images/capi/packer/nutanix/rhel-9.json      | 2 +-
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/images/capi/packer/nutanix/packer.json.tmpl b/images/capi/packer/nutanix/packer.json.tmpl
index f53c51514..3d4e875b0 100644
--- a/images/capi/packer/nutanix/packer.json.tmpl
+++ b/images/capi/packer/nutanix/packer.json.tmpl
@@ -132,6 +132,7 @@
     "image_delete": "false",
     "image_export": "false",
     "image_name": "{{user `build_name`}}-kube-{{user `kubernetes_semver`}}",
+    "image_url": "{{env `RHEL_IMAGE_URL`}}",
     "kubernetes_cni_deb_version": null,
     "kubernetes_cni_http_source": null,
     "kubernetes_cni_semver": null,
@@ -160,8 +161,8 @@
     "nutanix_username": "{{env `NUTANIX_USERNAME`}}",
     "python_path": "",
     "scp_extra_vars": "",
-    "source_image_delete": "false",
-    "source_image_force": "false",
+    "source_image_delete": "true",
+    "source_image_force": "true",
     "ssh_password": "$SSH_PASSWORD",
     "ssh_username": "builder",
     "vm_force_delete": "true"
diff --git a/images/capi/packer/nutanix/rhel-8.json b/images/capi/packer/nutanix/rhel-8.json
index 9aba21d66..718984aef 100644
--- a/images/capi/packer/nutanix/rhel-8.json
+++ b/images/capi/packer/nutanix/rhel-8.json
@@ -5,8 +5,8 @@
   "epel_rpm_gpg_key": "https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-8",
   "extra_rpms": "",
   "guest_os_type": "Linux",
-  "image_url": "https://REPLACE_YOUR_SERVER/redhat/8/rhel-8.8-x86_64-kvm.qcow2",
+  "image_url": "{{user `image_url`}}",
   "redhat_epel_rpm": "https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm",
   "shutdown_command": "shutdown -P now",
   "user_data": "I2Nsb3VkLWNvbmZpZwp1c2VyczoKICAtIG5hbWU6IGJ1aWxkZXIKICAgIHN1ZG86IFsnQUxMPShBTEwpIE5PUEFTU1dEOkFMTCddCmNocGFzc3dkOgogIGxpc3Q6IHwKICAgIGJ1aWxkZXI6YnVpbGRlcgogIGV4cGlyZTogRmFsc2UKc3NoX3B3YXV0aDogVHJ1ZQ=="
 }
diff --git a/images/capi/packer/nutanix/rhel-9.json b/images/capi/packer/nutanix/rhel-9.json
index b7dddb4f2..c49a1a656 100644
--- a/images/capi/packer/nutanix/rhel-9.json
+++ b/images/capi/packer/nutanix/rhel-9.json
@@ -5,8 +5,8 @@
   "epel_rpm_gpg_key": "https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-9",
   "extra_rpms": "",
   "guest_os_type": "Linux",
-  "image_url": "https://REPLACE_YOUR_SERVER/redhat/9/rhel-9.2-x86_64-kvm.qcow2",
+  "image_url": "{{user `image_url`}}",
   "redhat_epel_rpm": "https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm",
   "shutdown_command": "shutdown -P now",
   "user_data": "I2Nsb3VkLWNvbmZpZwp1c2VyczoKICAtIG5hbWU6IGJ1aWxkZXIKICAgIHN1ZG86IFsnQUxMPShBTEwpIE5PUEFTU1dEOkFMTCddCmNocGFzc3dkOgogIGxpc3Q6IHwKICAgIGJ1aWxkZXI6YnVpbGRlcgogIGV4cGlyZTogRmFsc2UKc3NoX3B3YXV0aDogVHJ1ZQ=="
 }
-- 
2.49.0

